src/metabase/task/delete_inactive_tables.clj

```diff
@@ -25,8 +25,12 @@
   (log/debug (trs "Cleaning up Inactive Tables older than {0} days" inactive-table-max-days))
   (task-history/with-task-history {:task "delete-inactive-tables"}
     (t2/delete! :model/Table
-                :active false
-                :updated_at [:< (t/minus (t/zoned-date-time) (t/days inactive-table-max-days))])))
+                :active     false
+                :updated_at [:< (t/minus (t/zoned-date-time) (t/days inactive-table-max-days))]
+                ;; do not delete tables that have active cards associated with them (#35615)
+                :id         [:not-in {:select-distinct [:table_id]
```

Comment:
an interesting performance analysis on various ways to do this https://explainextended.com/2009/09/18/not-in-vs-not-exists-vs-left-join-is-null-mysql/